cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -70,6 +71,19 @@ func CreateInitKubeConfigFiles(outDir string, cfg *kubeadmapi.MasterConfiguratio
 	)
 }
 
+// CreateJoinMasterKubeConfigFiles will create and write to disk the kubeconfig files required by kubeadm
+// join --master workflow, plus a the admin kubeconfig file to be deployed on the new master.
+// If kubeconfig files already exists, they are used only if evaluated equal; otherwise an error is returned.
+func CreateJoinMasterKubeConfigFiles(outDir string, cfg *kubeadmapi.MasterConfiguration) error {
```

Comment:
do we need an entirely new fn, or can we refactor CreateInitKubeConfigFiles to, for example, accept a new arg?

Reply:
I prefer two functions instead of a parameter that the somehow will mix-up the concerns of the join --master workflow into the init workflow.
However I'm open to change this if necessary